text/0017-gas-futures.md

```diff
@@ -0,0 +1,144 @@
+
+<!-- ```python -->
+<!-- FUTURES_GAS_LIMIT[i] = BLOCK_GAS_LIMIT * FUTURES_SOLD[i] * BST_SIZE / GAS_PER_DAY -->
+<!-- ``` -->
+
+<!-- **Example:** To determine `FUTURES_QUOTA`, we work our way back from the `FUTURES_GAS_LIMIT` we desire. Let us assume that we want 20% of block space to be reserved for futures transactions. This sets our upper limit in the case which the whole quota is sold off, i.e. `FUTURES_SOLD[i] = FUTURES_QUOTA`. Solving the equation with these values, we obtain `FUTURES_QUOTA = 0.2 * GAS_PER_DAY / BST_SIZE`. Assuming `2**14 = 16384 ms` rounds, we have `60*60*24/16.384 ~ 5273` blocks per day, giving us `GAS_PER_DAY = 5.273e15`. Finally, we compute `FUTURES_QUOTA = 0.2 * 5.273e15 / 1e9 = 1054600` BST per day. -->
+
+### Auction
+
+The auction for `GAS-YYYY-MM-DD-NNNNN` begins at tick `X - GAS_AUCTION_PERIOD * TICKS_IN_DAY` where `X` is the tick corresponding to `YYYY-MM-DD` 00:00 in UTC, and `TICKS_IN_DAY = 24*60*60*1000` is the number of ticks in one day. The auction can go on until the whole quota is sold off.
+
+The auction price is determined by the validators collectively, who announce the price they are willing to accept. The futures price is computed by taking a weighted average:
+
+```python
+auction_price = sum(weight[v] * price[i] for v in validators) / sum(weight[v] for v in validators)
+```
+
+where `weight[v]` is the consensus weight of and `price[v]` is the price announced by validator `v`.
+
+**Note:** `auction_price` designates the **gas price** that the user will receive on the day of delivery, not the price per BST.
+
+Preferred prices for a given futures can be announced prior to and during the auction by any validator. Auctions begin automatically without any outside intervention, and if a validator does not announce a price for the new futures, the price from the previous auction is assumed. Finally, the very first auction starts off with the preferred price set to zero for all validators.
+
+### Payment and delivery
+
+When a user buys a certain `amount` of the futures `BST-YYYY-MM-DD-NNNNN`, they make an upfront payment of
+
+```python
+escrow = auction_price * amount * BST_SIZE # CSPR
+```
+
+which is placed in escrow, until the date of delivery. This mints a [non-fungible token](https://en.wikipedia.org/wiki/Non-fungible_token) (NFT) which we call the *futures escrow token* (FET). A FET represents a user's ownership of the BST, and stores the information
+
+```python
+(
+    owner, # The user who bought the futures
+    symbol, # BST symbol e.g. BST-2020-09-01-01000. Can also be any type of reference.
+    amount, # The amount of futures bought
+    price, # The auction price at the time of purchase
+    total_gas_used, # Amount of gas used with this FET
```

Comment:
I don't think we need that anymore, since the whole FET now refers to only a single block?